pkg/query-service/app/queryBuilder/query_builder.go

```diff
@@ -383,6 +387,10 @@ func (c *cacheKeyGenerator) GenerateKeys(params *v3.QueryRangeParamsV3) map[stri
 		} else if query.Expression == queryName && query.DataSource == v3.DataSourceMetrics {
 			var parts []string
 
+			if params.Version != "v4" && params.CompositeQuery.PanelType != v3.PanelTypeGraph {
```

Comment:
please add a code comment here stating that, for metring if the version is v3 then we only support caching for Graph.

Will be easier to look back

Reply:
Added in https://github.com/SigNoz/signoz/pull/5651/commits/6d845923899135c4f7df6a161164deaf2ddb717b